examples/cpp/dds/RequestReplyExample/CalculatorClient.cpp

```diff
@@ -0,0 +1,288 @@
+            reply_topic_ = nullptr;
+        }
+
+        if (nullptr != request_topic_)
+        {
+            participant_->delete_topic(request_topic_);
+            request_topic_ = nullptr;
+        }
+
+        if ( nullptr != subscriber_)
+        {
+            participant_->delete_subscriber(subscriber_);
+            subscriber_ = nullptr;
+        }
+
+        if (nullptr != publisher_)
+        {
+            participant_->delete_publisher(publisher_);
+            publisher_ = nullptr;
+        }
+
+        if (nullptr != participant_)
+        {
+            eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->delete_participant(participant_);
+            participant_ = nullptr;
+        }
+    }
+
+    int64_t request(
+            OperationType operation,
+            int32_t x,
+            int32_t y)
+    {
+        int64_t z = 0;
+        RequestType request;
+        request.operation(operation);
+        request.x(x);
+        request.y(y);
+
+        if (request_writer_->write(static_cast<void*>(&request), listener_.write_params))
+        {
+            std::unique_lock<std::mutex> lock(listener_.reception_mutex);
+            listener_.reception_cv.wait(lock, [&]()
+                    {
+                        return listener_.received_reply;
+                    });
+            z = listener_.z;
+        }
+        else
+        {
+            std::cerr << "Error writing the request" << std::endl;
+        }
+
+        return z;
+    }
+
+private:
+
+    eprosima::fastdds::dds::DomainParticipant* participant_ = nullptr;
+
+    eprosima::fastdds::dds::Publisher* publisher_ = nullptr;
+
+    eprosima::fastdds::dds::Subscriber* subscriber_ = nullptr;
+
+    eprosima::fastdds::dds::Topic* request_topic_ = nullptr;
+
+    eprosima::fastdds::dds::Topic* reply_topic_ = nullptr;
+
+    eprosima::fastdds::dds::DataWriter* request_writer_ = nullptr;
+
+    eprosima::fastdds::dds::DataReader* reply_reader_ = nullptr;
+
+    eprosima::fastdds::dds::TypeSupport request_type_;
+
+    eprosima::fastdds::dds::TypeSupport reply_type_;
+};
+
+void print_help()
+{
+    std::cout << "Usage: CalculatorClient <x> <+|-|x|/> <y>" << std::endl;
+}
+
+int main(
+        int argc,
+        char** argv)
+{
+    argc -= (argc > 0); argv += (argc > 0); // skip program name argv[0] if present
+
+    if (3 != argc)
+    {
+        print_help();
+        return -1;
+    }
+
+    int32_t x;
+    int32_t y;
+    OperationType operation;
+    char* endptr = 0;
+
+    x = strtol(argv[0], &endptr, 10);
+    if (endptr == argv[0] || *endptr != 0)
+    {
+        std::cerr << "Error reading numeric argument x." << std::endl;
+        print_help();
+        return -1;
+    }
+
+    if (0 == strcmp(argv[1], "+"))
+    {
+        operation = OperationType::ADDITION;
+    }
+    else if (0 == strcmp(argv[1], "-"))
+    {
+        operation = OperationType::SUBSTRACTION;
+    }
+    else if (0 == strcmp(argv[1], "x"))
```

Comment:
I think it is more standard to use `*` for multiplication when talking about a program (or maybe it is my programming bias). Reading [wikipedia](https://en.wikipedia.org/wiki/Multiplication_sign) it seems like considering `x` (lowercase x) as the multiplication sign is considered incorrect in mathematical writing. So I suppose it is better to follow the ASCII convention and use `*`.

Reply:
Linux shells substitute `*` for all current files in the current directory. This is the reason I've select `x`.

Reply:
Fair enough!